client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -1391,10 +1391,16 @@ public void onSuccess(ByteBuffer response) {
                 groupedBatchId,
                 Arrays.toString(batchIds));
             pushState.removeBatch(groupedBatchId, hostPort);
-            if (response.remaining() > 0 && response.get() == StatusCode.MAP_ENDED.getValue()) {
-              mapperEndMap
-                  .computeIfAbsent(shuffleId, (id) -> ConcurrentHashMap.newKeySet())
-                  .add(mapId);
+            if (response.remaining() > 0) {
+              int retCode = response.get();
+              if (retCode == StatusCode.MAP_ENDED.getValue()) {
```

Comment:
Is there any compatibility problem for old client version and new server version?

Reply:
This is possible. I'll fix it.